ansible_collections/arista/avd/roles/eos_designs/templates/facts/switch/switch.j2

```diff
@@ -450,55 +450,60 @@ switch:
 {%     set leaf.vlans = []  %}
 {%     set leaf.tenants = {} %}
 {%     set leaf.pseudowires = [] %}
-{%     for tenant in tenants | arista.avd.natural_sort if tenant in switch_filter_tenants or "all" in switch_filter_tenants %}
-{%         set add_vrfs = {} %}
-{%         for vrf in tenants[tenant].vrfs | arista.avd.natural_sort %}
-{%             set add_svis = [] %}
-{%             for svi in tenants[tenant].vrfs[vrf].svis | arista.avd.natural_sort %}
-{%                 for svi_tag in tenants[tenant].vrfs[vrf].svis[svi].tags | arista.avd.natural_sort %}
-{%                     if svi_tag in switch_filter_tags or switch_data.group is arista.avd.defined(svi_tag) or "all" in switch_filter_tags %}
-{%                         do add_svis.append(svi) %}
-{%                         do leaf.vlans.append(svi | int) %}
-{%                         break %}
-{%                     endif %}
-{%                 endfor %}
-{%             endfor %}
+{%     set tmp_hostvars = hostvars[inventory_hostname] %}
+{%     for network_services_key in network_services_keys | arista.avd.natural_sort('name') %}
+{%         if network_services_key.name is arista.avd.defined and tmp_hostvars[network_services_key.name] is arista.avd.defined %}
+{%             for tenant in tmp_hostvars[network_services_key.name] | arista.avd.convert_dicts | arista.avd.natural_sort('name') if tenant.name in switch_filter_tenants or "all" in switch_filter_tenants %}
+{%                 set add_vrfs = {} %}
+{%                 for vrf in tenant.vrfs | arista.avd.convert_dicts | arista.avd.natural_sort('name') %}
+{%                     set add_svis = [] %}
+{%                     for svi in vrf.svis | arista.avd.convert_dicts('id') | arista.avd.natural_sort('id') %}
+{%                         for svi_tag in svi.tags | arista.avd.natural_sort %}
+{%                             if svi_tag in switch_filter_tags or switch_data.group is arista.avd.defined(svi_tag) or "all" in switch_filter_tags %}
```

Comment:
Looks good to me. Just curious to know if we can use ```switch_filter_tags is arista.avd.contains([svi_tag, 'all'])``` instead of ```svi_tag in switch_filter_tags``` and ```"all" in switch_filter_tags```

Reply:
Same for line 456, ```if tenant.name in switch_filter_tenants or "all" in switch_filter_tenants```

Reply:
Those are part of another PR, so this will come later